src/ai/core/signal_processor.py

```
@@ -0,0 +1,172 @@
+
+        return {
+            'final_score': final_score,
+            'component_scores': {
+                'intent_score': intent_score,
+                'market_score': market_score,
+                'regional_score': regional_score,
+                'trend_score': trend_score
+            }
+        }
+
+    def generate_match_reasons(self, component_scores: Dict) -> List[str]:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)